app/src/main/kotlin/no/nav/tiltakspenger/vedtak/service/behandling/BehandlingServiceImpl.kt

```diff
@@ -88,12 +96,7 @@ class BehandlingServiceImpl(
 
         when (behandling) {
             is BehandlingTilBeslutter -> {
-                sessionOf(DataSource.hikariDataSource).use {
-                    it.transaction { txSession ->
-                        behandlingRepo.lagre(behandling.sendTilbake(utøvendeBeslutter), txSession)
-                        attesteringRepo.lagre(attestering, txSession)
-                    }
-                }
+                multiRepo.lagre(behandling.sendTilbake(utøvendeBeslutter), attestering)
```

Comment:
Dette var en stor forbedring! Bra å bli kvitt det "tekniske" rundt det som er viktig. Men, jeg skjønner fortsatt ikke helt hva den resulterende kodelinja gjør. Er det mulig å gjøre den enda mer lesbar? 

Prøver å utdype litt:
Er det én "operasjon" vi gjør her, eller gjør vi flere ting? I og med vi gjør både behandling.sendTilbake og multiRepo.lagre med returverdien av `sendTilbake` på ett og samme sted, lurer jeg på om de er to steg i å få utført samme operasjon (muligens i å sende tilbake en behandling til en saksbehandler som har vært hos beslutter)? Går det an å putte alt i én funksjon, f.eks. `sendBehandlingTilbakeTilSaksbehandler`?

Reply:
Jeg skjønner at denne ikke er så lett å forstå. Det er et forsøk på å bli kvitt transaksjonskoden fra service-laget, for å gjøre servicen mer testbar. Greia er at man ønsker å lagre tre ting i samme transaksjon, men også at kallet til utbetaling skal være i samme transaksjon - slik at hvis kallet til utbetaling feiler, så skal ikke database-lagringen ble gjennomført. Dette er nok midlertidig, for det kallet til utbetaling trengs ikke egentlig lenger. Så det kommer til å bli bedre snart! :)